images/virtualization-artifact/pkg/controller/vd/internal/life_cycle.go

```diff
@@ -90,9 +90,29 @@ func (h LifeCycleHandler) Handle(ctx context.Context, vd *virtv2.VirtualDisk) (r
 			return reconcile.Result{}, fmt.Errorf("failed to clean up to restart import process: %w", err)
 		}
 
+		vd.Status.StorageClassName = ""
+
 		return reconcile.Result{Requeue: true}, nil
 	}
 
+	storageClassReadyCondition, ok := service.GetCondition(vdcondition.StorageClassReadyType, vd.Status.Conditions)
+	if !ok {
+		return reconcile.Result{Requeue: true}, fmt.Errorf("condition %s not found", vdcondition.StorageClassReadyType)
+	}
+
+	if readyCondition.Status != metav1.ConditionTrue && storageClassReadyCondition.Status != metav1.ConditionTrue && vd.Status.StorageClassName != "" {
```

Comment:
How about to unite conditions from line 79 and 103? Looks like they do the same: clean up